### ISSUE TITLE
 guard-for-in is unable to detect if statement guarding for...in loop

### ISSUE THREAD

**Issue by Yangfan2016** [eslint/eslint#11926]
**Tell us about your environment**

* **ESLint Version:**
5.16.0
* **Node Version:**
10.16.0
* **npm Version:**
6.9.0
**What parser (default, Babel-ESLint, etc.) are you using?**

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  env: {
    browser: true,
    es6: true,
  },
  extends: [
    'airbnb-base',
  ],
  globals: {
    Atomics: 'readonly',
    SharedArrayBuffer: 'readonly',
  },
  parserOptions: {
    ecmaVersion: 2018,
    sourceType: 'module',
  },
  rules: {
    'no-continue': 0,
    'no-plusplus': 0,
    'no-unused-expressions': [
      2,
      {
        'allowShortCircuit': true, // 允许 短路求值
        'allowTernary': true // 允许 三目运算符
      }
    ]
  },
};

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
class FormValidator extends Schema {
  constructor(formEl, rule, cb) {
    const r = { ...defaultRule, ...rule };
    super(r);

    cb && (this.cb = cb);

    const form = document.querySelector(formEl);
    formItems = form.querySelectorAll('[data-prop]');
    // 绑定事件
    for (const field in r) {
      if (!Object.prototype.hasOwnProperty.call(r, field)) continue;
      let rules = r[field];
      rules = rules instanceof Array ? rules : [rules];
      const triggerRules = rules.filter(item => {
        return item.trigger !== void 0;
      });
      if (triggerRules.length === 0) continue;
      for (let i = 0, len = triggerRules.length; i < len; i++) {
        const tr = triggerRules[i];
        const curEl = [...formItems].find(item => {
          return item.dataset.prop === field;
        });
        curEl.addEventListener(tr.trigger, ev => {
          const { value } = ev.target;
          const model = {
            [field]: value,
          };
          let s = new Schema({
            [field]: tr,
          });
          s.validate(model, (err, fields) => {
            this.finalCallback(err, fields);
          });
          s = null;
        });
      }
    }
  }
}
```

```bash
eslint
```

**What did you expect to happen?**

No error

**What actually happened? Please include the actual, raw output from ESLint.**
for..in loops iterate over the entire prototype chain, which is virtually never what you want. Use Object.{keys,values,entries}, and iterate over the resulting array

**Are you willing to submit a pull request to fix this bug?**

**Comment by g-plane** [comment]
Hey @Yangfan2016 , thanks for the issue.

It seems that I can't reproduce this: https://eslint.org/demo/#eyJ0ZXh0IjoiZm9yIChjb25zdCBmaWVsZCBpbiByKSB7XHJcbiAgICBpZiAoIU9iamVjdC5wcm90b3R5cGUuaGFzT3duUHJvcGVydHkuY2FsbChyLCBmaWVsZCkpIGNvbnRpbnVlO1xyXG4gICAgbGV0IHJ1bGVzID0gcltmaWVsZF07XHJcbiAgICAvLyBSZXN0IG9mIGNvZGUuLi5cclxufSIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6Niwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6eyJndWFyZC1mb3ItaW4iOjJ9LCJlbnYiOnt9fX0= .

**Comment by Yangfan2016** [comment]
It still dosen't work with my code (eslint@5.16.0 )


![image](https://user-images.githubusercontent.com/17830872/60409031-472d7e00-9bf4-11e9-8f59-4584bb158456.png)

**Comment by g-plane** [comment]
Can you simplify your code to make it clear?

**Comment by kaicataldo** [comment]
@Yangfan2016 looking at the screenshot you provided, it looks like this is coming from no-restricted-syntax, and I’m guessing is coming from the airbnb configuration you’re extending off of.

**Comment by kaicataldo** [comment]
[Here](https://github.com/airbnb/javascript/blob/b85baeafed8b66fdd9756439a0b8774860147913/packages/eslint-config-airbnb-base/rules/style.js#L334) it is. If you believe there is a bug in the rule, feel free to make an issue on that repo. That being said, it looks like the intention here is to not allow that syntax at all, so I believe it’s working correctly.

**Comment by Yangfan2016** [comment]
thx, It works @kaicataldo
